bahmnicore-omod/src/main/resources/liquibase.xml

```diff
@@ -4658,13 +4673,143 @@
+
+            UPDATE concept set version='BAHMNI-CORE-1.0' where concept_id=@concept_id;
+        </sql>
+    </changeSet>
+
+    <changeSet id="bahmni-emrapi-202208181243" author="BAHMNI" runAlways="true">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="1">
+                select count(*) from concept_reference_map rmap
+                join concept_reference_term rterm on rmap.concept_reference_term_id = rterm.concept_reference_term_id
+                and rterm.code='Diagnosis Concept Set'
+                join concept_reference_source rsource on rsource.concept_source_id = rterm.concept_source_id and rsource.name
+                = 'org.openmrs.module.emrapi'
+                join concept c on c.concept_id = rmap.concept_id and c.version is null ;
+            </sqlCheck>
+        </preConditions>
+        <comment> Retire non Bahmni Diagnosis concept with reference mapping code 'Diagnosis Concept Set' </comment>
+        <sql>
+            SET @concept_source_id = 0;
+            SET @concept_reference_term_id = 0;
+            SET @concept_id = 0;
+
+            SELECT concept_source_id into @concept_source_id from concept_reference_source where name = 'org.openmrs.module.emrapi';
+            SELECT concept_reference_term_id into @concept_reference_term_id from concept_reference_term where concept_source_id = @concept_source_id and code = 'Diagnosis Concept Set';
+            SELECT concept_id into @concept_id from concept_reference_map where concept_reference_term_id = @concept_reference_term_id order by date_created desc limit 1;
+
+            UPDATE concept set retired = TRUE where version is null and concept_id = @concept_id;
+        </sql>
+    </changeSet>
+
+    <changeSet id="bahmni-emrapi-202208181245" author="BAHMNI" runAlways="true">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="1">
+                select count(*) from concept_reference_map rmap
+                join concept_reference_term rterm on rmap.concept_reference_term_id = rterm.concept_reference_term_id
+                and rterm.code='Coded Diagnosis'
+                join concept_reference_source rsource on rsource.concept_source_id = rterm.concept_source_id and rsource.name
+                = 'org.openmrs.module.emrapi'
+                join concept c on c.concept_id = rmap.concept_id and c.version is null ;
+            </sqlCheck>
+        </preConditions>
+        <comment>Retire non Bahmni Diagnosis concept with reference mapping code 'Coded Diagnosis'</comment>
+        <sql>
+            SET @concept_source_id = 0;
+            SET @concept_reference_term_id = 0;
+            SET @concept_id = 0;
+
+            SELECT concept_source_id into @concept_source_id from concept_reference_source where name =
+            'org.openmrs.module.emrapi';
+            SELECT concept_reference_term_id into @concept_reference_term_id from concept_reference_term where
```

Comment:
@angshu relationship type check is not there in the existing emrapi API logic. This may not give the duplicate concept here but the API would again fail with multiple concepts if we have the check here. The logic that used is similar to what we have in the emrapi  API